datamol/viz/_lasso_highlight.py

```diff
@@ -0,0 +1,439 @@
+            canvas.DrawLine(Point2D(*atom_i_left_at), Point2D(*atom_j_right_at))
+            canvas.DrawLine(Point2D(*atom_i_right_at), Point2D(*atom_j_left_at))
+    # restoring prior line width
+    canvas.SetLineWidth(prior_lw)
+
+
+# TODO switch this over to other doc string
+def _draw_multi_matches(
+    canvas: rdMolDraw2D.MolDraw2D,
+    mol: Chem.Mol,
+    indices_set_lists: List[Union[list, str]],
+    r_min: float = 0.3,
+    r_dist: float = 0.13,
+    relative_bond_width: float = 0.5,
+    color_list: List[ColorTuple] = None,
+    line_width: int = 2,
+) -> None:
+    """
+    Parameters
+    ----------
+    canvas : rdMolDraw2D.MolDraw2D
+        RDKit Canvas, where highlighting is drawn to.
+    mol: Chem.Mol
+        Atoms from the molecule `mol` are takes as positional reference for the highlighting.
+    indices_set_lists: List[Union[list, str]]
+        Atom indices delineating highlighted substructure.
+    r_min: float
+        Radius of the smallest circle around atoms. Length is relative to average bond length (1 = avg. bond len).
+    r_dist: float
+        Incremental increase of radius for the next substructure.
+    relative_bond_width: float
+        Distance of line to "bond" (line segment between the two atoms). Size is relative to `atom_radius`.
+    line_width: int
+        width of drawn lines.
+    color_list: List[ColorTuple]
+        List of tuples with RGBA or RGB values specifying the color of the highlighting.
+    Returns
+    -------
+    None
+    Returns
+    -------
+    """
+    # If no colors are given, all substructures are depicted in gray.
+    if color_list is None:
+        color_list = [(0.5, 0.5, 0.5)] * len(indices_set_lists)
+    if len(color_list) < len(indices_set_lists):
+        raise ValueError("Not enough colors for referenced substructures!")
+
+    level_manager = defaultdict(set)
+    for match_atoms, color in zip(indices_set_lists, color_list):
+        used_levels = set.union(*[level_manager[a] for a in match_atoms])
+        if len(used_levels) == 0:
+            free_levels = {0}
+        else:
+            max_level = max(used_levels)
+            free_levels = set(range(max_level)) - used_levels
+
+        if free_levels:
+            draw_level = min(free_levels)
+        else:
+            draw_level = max(used_levels) + 1
+
+        for a in match_atoms:
+            level_manager[a].add(draw_level)
+
+        ar = r_min + r_dist * draw_level
+        _draw_substructurematch(
+            canvas,
+            mol,
+            match_atoms,
+            rel_radius=ar,
+            rel_width=max(relative_bond_width, ar),
+            color=color,
+            line_width=line_width,
+        )
+
+
+color_dict = {
+    "red": (1, 0, 0, 1),
+    "blue": (0, 0.5, 1, 1),
+    "orange": (1, 0.5, 0, 1),
+    "green": (0, 1, 0, 1),
+    "yellow": (1, 1, 0, 1),
+    "dark blue": (0, 0, 0.5, 1),
+}
+
+
+def lasso_highlight_image(
+    canvas_width: int,
+    canvas_height: int,
```

Comment:
I suggest using the same signature as in `dm.to_image()`: `mol_size: Union[Tuple[int, int], int] = (300, 300),` instead (and also move them last in the signature).